assets/opstools/BuildApp/controllers/data_fields/equation.js

```diff
@@ -0,0 +1,485 @@
+				typeRounding : $$(componentIds.typeRounding).getValue(),
+				typeThousands : $$(componentIds.typeThousands).getValue(),
+				typeFormat : $$(componentIds.typeFormat).getValue(),
+				template:'<div class="ab-equation-data-field"></div>',
+				
+			}
+		};*/
+	};
+
+
+	equationDataField.customDisplay = function (application, object, fieldData, rowData, data, viewId, itemNode, options) {
+		/*
+		if (rowData == null) {
+			$(itemNode).find('.ab-equation-data-field').html('');
+			return true;
+		}
+		
+		var decimalSizeNum = 0,
+			decimalDelimiters = ".",
+			groupDelimiters = "";
+		
+		if (fieldData.setting.typeDecimals && fieldData.setting.typeDecimals != 'none') {
+			if (fieldData.setting.typeDecimalPlaces != undefined && fieldData.setting.typeDecimalPlaces != 'none') {
+				decimalSizeNum = fieldData.setting.typeDecimalPlaces;
+			}
+			if (fieldData.setting.typeDecimals != undefined) {
+				switch (fieldData.setting.typeDecimals) {
+					case 'period':
+						decimalDelimiters = ".";
+						break;
+					case 'comma':
+						decimalDelimiters = ",";
+						break;
+				}
+			}*/
+
+			/*if (fieldData.setting.typeRounding != undefined) {
+				switch (fieldData.setting.typeRounding) {
+					case 'roundUp':
+						var num = data;
+						var precision = -decimalSizeNum;
+						var div = Math.pow(10, precision);
+						data = Math.ceil(num / div) * div;
+						break;
+					case 'roundDown':
+						var num = data;
+						var precision = -decimalSizeNum;
+						var div = Math.pow(10, precision);
+						data = Math.floor(num / div) * div;
+						break;
+				}
+			}*/
+		//}
+		
+		/*if (fieldData.setting.typeThousands != undefined) {
+			switch (fieldData.setting.typeThousands) {
+				case 'comma':
+					groupDelimiters = ",";
+					break;
+				case 'period':
+					groupDelimiters = ".";
+					break;
+				case 'space':
+					groupDelimiters = " ";
+					break;
+			}
+		}*/
+		
+		/*var data = caldateDiff(fieldData.setting.dateType,rowData[fieldData.setting.equation],new Date());
+		
+		console.log("datakid: "+ data);
+		
+		var numberFormat = webix.Number.format(data, {
+			groupDelimiter: groupDelimiters,
+			groupSize: 3,
+			decimalDelimiter: decimalDelimiters,
+			decimalSize: decimalSizeNum
+		});*/
+		
+		//console.log("numberFormatkid: "+ numberFormat);
+		/*if (fieldData.setting.typeFormat != undefined && fieldData.setting.typeFormat != 'none') {
+			var formatItem = formatList.find(function (item) { return item.id == fieldData.setting.typeFormat });
+			if (formatItem) {
+				numberFormat = (formatItem.position == 'prefix' ? formatItem.sign + ' ' + numberFormat : numberFormat + ' ' + formatItem.sign);
+			}
+		}*/
+		
+		//$(itemNode).find('.ab-equation-data-field').html(numberFormat);
+		
+		if (rowData == null) {
+			$(itemNode).find('.ab-equation-data-field').html('');
+			return true;
+		}
+	
+		try {
+
+			var value = FunctionManager.parseEquation(fieldData.setting.equation, rowData);
```

Comment:
instead of this try{}catch(){}  
do something like this:

```javascript
var parser = EquationManager.parse(fieldData.setting.equation);
if (parser) {

	var value = parser(rowData);

	var resultType = this.getDataField(fieldData.setting.equationType);
	if(resultType.customDisplay) {
		resultType.customDisplay(application, object, { setting: fieldData.setting.resultSettings }, rowData, value, viewId, itemNode, options);
	} else {
		// else do whatever we have to do to display this value
		$(itemNode).find('.ab-equation-data-field').html(value);
	}
} else {
	$(itemNode).find('.ab-equation-data-field').html('invalid equation!');
}
```